lib/rubocop/cop/style/unless_logical_operators.rb

```diff
@@ -0,0 +1,79 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # This cop checks for the use of logical operators in an unless condition.
+      #
+      # @example EnforcedStyle: forbid_mixed_logical_operator (default)
+      #   # bad
+      #   return unless a || b && c
+      #   return unless a && b || c
+      #   return unless a && b and c
+      #   return unless a || b or c
+      #   return unless a && b or c
+      #   return unless a || b and c
+      #
+      #   # good
+      #   return unless a && b && c
+      #   return unless a || b || c
+      #   return unless a and b and c
+      #   return unless a or b or c
+      #   return unless a?
+      #
+      # @example EnforcedStyle: forbid_logical_operator
+      #   # bad
+      #   return unless a || b
+      #   return unless a && b
+      #   return unless a or b
+      #   return unless a and b
+      #
+      #   # good
+      #   return unless a
+      #   return unless a?
+      class UnlessLogicalOperators < Base
+        include ConfigurableEnforcedStyle
+
+        FORBID_MIXED_LOGICAL_OPERATOR = 'Do not use mixed logical operators in an unless condition.'
+        FORBID_LOGICAL_OPERATOR = 'Do not use a logical operator in an unless condition.'
+
+        def_node_matcher :or_with_and?, <<~PATTERN
+          (if (or <`and ...> ) ...)
+        PATTERN
+        def_node_matcher :and_with_or?, <<~PATTERN
+          (if (and <`or ...> ) ...)
+        PATTERN
+        def_node_matcher :logical_operator?, <<~PATTERN
+          (if ({and or} ... ) ...)
+        PATTERN
+
+        def on_if(node)
+          return unless node.unless?
+
+          if style == :forbid_mixed_logical_operator && mixed_logical_operator?(node)
+            add_offense(node, message: FORBID_MIXED_LOGICAL_OPERATOR)
+          elsif style == :forbid_logical_operator && logical_operator?(node)
+            add_offense(node, message: FORBID_LOGICAL_OPERATOR)
+          end
+        end
+
+        private
+
+        def mixed_logical_operator?(node)
+          or_with_and?(node) ||
+            and_with_or?(node) ||
+            mixed_precedence_and?(node) ||
+            mixed_precedence_or?(node)
+        end
+
+        def mixed_precedence_and?(node)
+          node.source.include?('&&') && node.source.include?('and')
```

Comment:
I'm not sure if there is a better way to distinguish `a && b && c` and `a && b and c`.

Using node matcher doesn't work because both have the following AST:
```
(and
  (and
    (send nil :a)
    (send nil :b))
  (send nil :c))
```

Same goes for `a || b || c` and `a || b or c`.

Reply:
I considered using a node matcher for `a and b && c`, using `(if (and ... (and ...)) ...)` to match the following AST:
```
(and
  (send nil :a)
  (and
    (send nil :b)
    (send nil :c)))
```

However, if we need to check the `node.source` anyway, then it's unnecessary to have the node matcher.

Reply:
That's ok. Technically you don't need to use `include?`, though as you can do an equality check instead (and I guess it'd be faster).

Reply:
Could you share an example how we can do an equality check on this?

Reply:
Isn't this just a regular string? How about `==`? :-)